#174 - Fixed bug resolving project paths with multiple peroids in project name

roketworks opened the pull request:
#174 - Proposed fix  

Thanks for contributing to the buildpack. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:
Moved the logic that gets the project path for the start command into the MainPath() function, allowing for different strategies to be applied for different project types.

* An explanation of the use cases your change solves
Allows more different logic when resolving the project path for different project types (prepublished etc.)

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [ ] I have added an integration test

roketworks commented:
Sent CLA Agreement to contributors@cloudfoundry.org

roketworks commented:
Reopening

dgodd commented:
Thanks @roketworks , unfortunately it appears that these changes have unfortunate consequences on pushing source code. I am currrently trying to work through all of the options.

roketworks commented:
@dgodd Thanks. I've moved it into the fix into `PublishStartCommand`, which I think is a more sensible place for it anyway. The unit tests are now reverted to expect all the same values as before the fix, so this should hopefully avoid any issues with pushing source, and contains the fix only the relevant function.

sclevine commented:
Closing this PR, as alternative changes are on `develop`.
